### PR TITLE
Prevent warnings from collection of TestFactories by unit testing frameworks

### PR DESCRIPTION
**Pull request by alexforencich**
Prevent pytest from issuing a PytestCollectionWarning when using TestFactory

### CODE REVIEW

**Comment by ktbarrett** [comment]
You are going to have to be far more descriptive than that. What does pytest have to do with cocotb? What are the details behind this bug?

**Comment by alexforencich** [comment]
When I run a cocotb testbench that imports `TestFactory` using `cocotb-test`, `pytest` tries to collect `TestFactory` since it starts with the word `Test` and since it has an `__init__` method it results in a `PytestCollectionWarning`.  The solution is to add `__test__ = False` inside `TestFactory` to prevent `pytest` from attempting to collect it.

File: cocotb/regression.py [hunk 1]
```diff
@@ -654,6 +654,9 @@ class TestFactory:
     the test description) includes the name and description of each generator.
     """
 
+    # Prevent PytestCollectionWarning
```
Comment (cmarqu) [inline cocotb/regression.py:657]:
```suggestion
    # Prevent PytestCollectionWarning - do not try to collect "TestFactory" despite it matching "Test*"
```

File: cocotb/regression.py [hunk 2]
```diff
@@ -654,6 +654,10 @@ class TestFactory:
     the test description) includes the name and description of each generator.
     """
 
+    # Prevent PytestCollectionWarning from pytest
```
Comment (ktbarrett) [inline cocotb/regression.py:657]:
```suggestion
    # Prevent PytestCollectionWarning from cocotb-test
```

**Review by ktbarrett — APPROVED** [review]
I'm fine with this. We do have to place nice with extensions.

File: cocotb/regression.py (hunk 2 above)
Comment (cmarqu) [reply cocotb/regression.py:657]: Strictly speaking, ``__test__`` also works with ``unittest`` and ``nose``.

Comment (ktbarrett) [reply cocotb/regression.py:657]: Then maybe something more general?
```suggestion
    # Prevent warnings from collection of TestFactories by unit testing frameworks.
```

Comment (alexforencich) [reply cocotb/regression.py]: Yeah, that's a much better description.

**Comment by themperek** [comment]
@alexforencich Thank you. First commit ✨ !

**Comment by cmarqu** [comment]
@alexforencich Thanks for this PR, and also all your other open source work!

**Pull request merged**